lib/breacan/client/files.rb

```diff
@@ -14,8 +15,35 @@ def files_list(args)
       end
 
       def files_upload(args)
-        # content_type = 'multipart/form-data'
-        # get 'files.upload', query: args
+        set_files_upload_builder
+        data = {}
+        if args[:file]
+          data[:file] = args[:file]
+          args.delete(:file)
+        end
+        data[:query] = args
+        res = post 'files.upload', data
+        reset_files_upload_builder
+        res
+      end
+
+      def set_files_upload_builder
+        @serializer = Breacan::Client::Files::FileSerializer.any_json
+        @middleware = Faraday::RackBuilder.new do |c|
+          c.request :multipart
+          c.request :url_encoded
+          c.adapter :net_http
+        end
+      end
+
+      def reset_files_upload_builder
+        @middleware = default_builder
+      end
+
+      class FileSerializer < Sawyer::Serializer
+        def encode(data)
+          data
```

Comment:
default serializer of saywer encode UploadIO object to String...

https://github.com/lostisland/sawyer/blob/master/lib/sawyer/serializer.rb#L49